lib/vector-common/src/byte_size_of.rs

```diff
@@ -129,3 +136,9 @@ impl ByteSizeOf for Value {
         }
     }
 }
+
+impl ByteSizeOf for DateTime<Utc> {
+    fn allocated_bytes(&self) -> usize {
+        0
```

Comment:
Is this right? I think we may need to sum up the fields.

Reply:
Well, I did some investigation about `DateTime<Utc>`. It's basically as follow
```rust
DateTime {
  datetime: NaiveDateTime {
    date: NaiveDate {
      ymdf: DateImpl, // i32
    },
    time: NaiveTime {
      secs: u32,
      frac: u32,
    },
  },
  offset: Utc, // an enum
}
```
So when I run the following piece of code
```rust
    let now: DateTime<Utc> = Utc::now(); 
    println!("datetime: {}", std::mem::size_of_val(&now)); // 12
    println!("utc: {}", std::mem::size_of::<Utc>()); // 0
    println!("i32: {}", std::mem::size_of::<i32>()); // 4
    println!("u32: {}", std::mem::size_of::<u32>()); // 4
```
I get `DateTime<Utc>` using 12 bytes in the end 🤔 
Please tell me if I'm forgetting something 😉

Reply:
Thanks for verifying, that does look right then 😄